Source/Fluids/WarpXFluidContainer.H

```diff
@@ -0,0 +1,172 @@
+/* Copyright 2023 Grant Johnson, Remi Lehe
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef WARPX_WarpXFluidContainer_H_
+#define WARPX_WarpXFluidContainer_H_
+
+#include "Evolve/WarpXDtType.H"
+#include "Initialization/PlasmaInjector.H"
+#include "MultiFluidContainer.H"
+
+#include<AMReX_MultiFab.H>
+#include<AMReX_Vector.H>
+
+#include <string>
+
+
+/**
+ * WarpXFluidContainer is the base polymorphic class from which all concrete
+ * fluid container classes derive.
+ *
+ * WarpXFluidContainer contains the main functions for initialization,
+ * interaction with the grid (field gather and current deposition), fluid
+ * source and push.
+ */
+class WarpXFluidContainer
+{
+public:
+    friend MultiFluidContainer;
+
+    WarpXFluidContainer (int nlevs_max, int ispecies, const std::string& name, const amrex::Geometry& geom);
+    ~WarpXFluidContainer() {}
+
+    void AllocateLevelMFs (int lev, const amrex::BoxArray& ba, const amrex::DistributionMapping& dm);
+
+    void InitData (int lev, amrex::Box init_box, amrex::Real cur_time);
+
+    void ReadParameters ();
+    /**
+     * Evolve updates a single timestep (dt) of the cold relativistic fluid eqautions
+     */
+    void Evolve (int lev,
+        const amrex::MultiFab& Ex, const amrex::MultiFab& Ey, const amrex::MultiFab& Ez,
+        const amrex::MultiFab& Bx, const amrex::MultiFab& By, const amrex::MultiFab& Bz,
+        amrex::MultiFab* rho, amrex::MultiFab& jx, amrex::MultiFab& jy, amrex::MultiFab& jz,
+        amrex::Real cur_time, bool skip_deposition=false);
+
+    /**
+     * AdvectivePush_Muscl takes a single timestep (dt) of the cold relativistic fluid equations
+     * using a Muscl-Handcock scheme
+     */
+    /**
```

Comment:
Should this be removed?

```suggestion
```

Reply:
Same comment for all the docstrings below.

Reply:
Completed.